src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionLiteral.cs

```diff
@@ -96,5 +141,92 @@ internal sealed partial class LocalRewriter
                 temp,
                 node.Type);
         }
+
+        private BoundExpression VisitListInterfaceCollectionLiteralExpression(BoundCollectionLiteralExpression node)
+        {
+            Debug.Assert(!_inExpressionLambda);
+            Debug.Assert(node.Type is { });
+
+            // PROTOTYPE: Improve perf. For instance, emit [] as Array.Empty<T>() rather than a List<T>.
+            var list = VisitCollectionInitializerCollectionLiteralExpression(node);
+            return _factory.Convert(node.Type, list);
+        }
+
+        private BoundExpression MakeCollectionLiteralSpreadElement(BoundCollectionLiteralSpreadElement initializer)
+        {
+            var enumeratorInfo = initializer.EnumeratorInfoOpt;
+            var addElementPlaceholder = initializer.AddElementPlaceholder;
+
+            Debug.Assert(enumeratorInfo is { });
+            Debug.Assert(addElementPlaceholder is { });
+            Debug.Assert(addElementPlaceholder.Type is { });
+
+            var syntax = (CSharpSyntaxNode)initializer.Syntax;
+            var iterationVariable = _factory.SynthesizedLocal(addElementPlaceholder.Type, syntax);
+            var convertedExpression = (BoundConversion)initializer.Expression;
+
+            AddPlaceholderReplacement(addElementPlaceholder, _factory.Local(iterationVariable));
+            var rewrittenAdd = VisitExpression(initializer.AddMethodInvocation);
+            Debug.Assert(rewrittenAdd is { });
+            var rewrittenBody = _factory.ExpressionStatement(rewrittenAdd);
+            RemovePlaceholderReplacement(addElementPlaceholder);
+
+            var elementPlaceholder = initializer.ElementPlaceholder;
+            var iterationVariables = ImmutableArray.Create(iterationVariable);
+            var breakLabel = new GeneratedLabelSymbol("break");
+            var continueLabel = new GeneratedLabelSymbol("continue");
+
+            BoundStatement statement;
+            if (convertedExpression.Operand.Type is ArrayTypeSymbol arrayType)
+            {
+                if (arrayType.IsSZArray)
+                {
+                    statement = RewriteSingleDimensionalArrayForEachEnumerator(
+                        initializer,
+                        convertedExpression.Operand,
+                        elementPlaceholder,
+                        elementConversion: null,
+                        iterationVariables,
+                        deconstruction: null,
+                        breakLabel,
+                        continueLabel,
+                        rewrittenBody);
+                }
+                else
```

Comment:
When is this true? I thought we filtered out converting to MD arrays in binding?

Reply:
This multi-dimensional array is the spread expression which is not necessarily a collection literal.
```
int[,] a = new[,] { { 1, 2 }, { 3, 4 } };
int[] b = [..a];
```